src/fabric/src/fabric_view_all_docs.erl

```diff
@@ -286,6 +286,20 @@ all_docs_concurrency() ->
             10
     end.
 
+validate_if_partition(Row, Acc) ->
+    QueryStr = Acc#vacc.req#httpd.qs,
+    case QueryStr of
+        undefined -> Row;
+        [{"partition", Id}] -> is_doc_in_partition(Row, Id)
+    end.
+is_doc_in_partition(Row, Id) ->
+    BinRowId = element(2, Row),
+    IsDocInPartition = binary:match(BinRowId, list_to_binary(Id)) /= nomatch,
+    case IsDocInPartition of
+        true -> Row;
+        false -> {view_row, BinRowId, undefined, undefined, undefined, undefined}
```

Comment:
this looks like you create a dummy row for entries that should not be in the result that presumably gets filtered out later. You also pass this into `Callback()` in line 318 below. Wouldn’t it make more sense to just not call `Callback()` with that row?

Reply:
and also it should be created as `#view_row{id = BinRowId}`

Reply:
👍